### PR TITLE
Eip747 implementation

### PR DESCRIPTION
**Pull request by chrismaree**
This PR implements [EIP747](https://ethereum-magicians.org/t/eip-747-wallet-watchasset/1048) which provides a simple mechanism to enable users to track tokens from their wallet. In the case of the voter dapp this will enable voters to track UMA tokens from their metamask wallet. For the sponsor dapp this will enable sponsors to add their newly minted tokens to their metamask wallet token tracker.

The screenshot below shows the metamask window.

image

Right now EIP747 does not provide any interface to detect if a token is already tracked. This makes the UX clunky for users that have already tracked a token. Ideally a "track token" button should only be shown if and only if user is not already tracking it. For this reason the track token button has been excluded from the voter dapp. When EIP747 provides a more responsive API for this interaction this will be updated. 

In the case of the sponsor dapp, at the completion of deployment and token creation the user is automatically prompted to track their new token.

### CODE REVIEW

File: sponsor-dapp-v2/src/components/Step6.js [hunk 1]
```diff
@@ -25,6 +25,21 @@ function Step6(props) {
...
+    params: {
+      type: "ERC20",
+      options: {
+        address: contractAddress,
+        symbol: symbol.substring(0, 6),
+        decimals: 18,
+        image: "https://umaproject.org/assets/images/UMA_square_red_logo.png"
+      }
+    },
+    id: Math.round(Math.random() * 100000)
```
Comment (mrice32) [inline sponsor-dapp-v2/src/components/Step6.js]: What's this id?

File: voter-dapp/src/TokenTracker.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  const addTokenToWallet = () => {
+    window.web3.currentProvider.sendAsync({
+      method: "wallet_watchAsset",
+      params: {
+        type: "ERC20",
+        options: {
+          address: drizzle.contracts.VotingToken.address,
+          symbol: "UMA",
+          decimals: 18,
+          image: "https://umaproject.org/assets/images/UMA_square_red_logo.png"
```
Comment (mrice32) [inline voter-dapp/src/TokenTracker.js]: @rcai1 @allisonlu22 do we want the UMA logo to show up for people's tokenbuilder tokens...?

File: sponsor-dapp-v2/src/components/Step3.js [hunk 3]
```diff
@@ -45,6 +45,7 @@ function useCreateContract(userSelectionsRef, identifierConfig, onNextStep, stat
       userSelectionsRef.current.contractAddress =
         TXObjects[TXObjects.length - 1].receipt.events.CreatedTokenizedDerivative.returnValues.contractAddress;
 
+      userSelectionsRef.current.symbol = state.tokenSymbol;
```
Comment (mrice32) [inline sponsor-dapp-v2/src/components/Step3.js]: Why was this necessary?

File: voter-dapp/src/TokenTracker.js (hunk 2 above)
Comment (rcai1) [reply voter-dapp/src/TokenTracker.js]: My preference is for the UMA logo itself to be reserved for the UMA token, rather than synthetic tokens that are created via the Synthetic Token Builder. 

We could create an alternative logo for synthetic tokens. Let me know if you'd like me to make a few mock-ups.

File: sponsor-dapp-v2/src/components/Step3.js (hunk 3 above)
Comment (mrice32) [reply sponsor-dapp-v2/src/components/Step3.js]: As discussed IRL, seems like the easiest way to move the symbol on to later stages.

File: voter-dapp/src/TokenTracker.js (hunk 2 above)
Comment (mrice32) [reply voter-dapp/src/TokenTracker.js]: Agreed, @rcai1 and @chrismaree, y'all can decide what logo (if any) to use here.

File: sponsor-dapp-v2/src/components/Step6.js (hunk 1 above)
Comment (mrice32) [reply sponsor-dapp-v2/src/components/Step6.js]: Checked out metamask's website, and it looks like this is the suggested way to do this. Feel free to disregard.

File: voter-dapp/src/TokenTracker.js (hunk 2 above)
Comment (chrismaree) [reply voter-dapp/src/TokenTracker.js]: I spent some time playing with different colors. 
UMA Token Icons options

**Comment by chrismaree** [comment]
PR on the website has been merged. Pop up looks like this now:
image
and once added to the wallet looks like this:
image
